Purchases/Caching/RCDeviceCache.m

```diff
@@ -109,6 +112,7 @@ - (void)clearCachesForAppUserID:(NSString *)oldAppUserID andSaveNewUserID:(NSStr
         [self deleteAttributesIfSyncedForAppUserID:oldAppUserID];
 
         [self cacheAppUserID:newUserID];
+        self.appUserIDHasBeenSet = YES;
```

Comment:
Should we move this to cacheAppUserId? That way if another function calls that function, we won't forget to set this bool

Reply:
yeah, good call. don't know what I was thinking.